api/server/router/volume/volume_routes.go

```diff
@@ -185,16 +185,18 @@ func handleVolAttachments(
 		return false
 	}
 
-	// determine a volume's attachment state
-	if len(vol.Attachments) == 0 {
-		vol.AttachmentState = types.VolumeAvailable
-	} else if iid != nil {
-		vol.AttachmentState = types.VolumeUnavailable
-		for _, a := range vol.Attachments {
-			if a.InstanceID != nil &&
-				strings.EqualFold(iid.ID, a.InstanceID.ID) {
-				vol.AttachmentState = types.VolumeAttached
-				break
+	// determine a volume's attachment state if it hasn't been set already
+	if vol.AttachmentState == 0 {
```

Comment:
Hi @codenrhoden,

I'll accept it if you fix this so that you avoid unnecessary indentation :) Let's change it to the following:

```go
	// determine a volume's attachment state if it hasn't been set already
	if vol.AttachmentState > 0 {
		return true
	}

	// determine a volume's attachment state
	if len(vol.Attachments) == 0 {
		vol.AttachmentState = types.VolumeAvailable
	} else if iid != nil {
		vol.AttachmentState = types.VolumeUnavailable
		for _, a := range vol.Attachments {
			if a.InstanceID != nil &&
				strings.EqualFold(iid.ID, a.InstanceID.ID) {
				vol.AttachmentState = types.VolumeAttached
				break
			}
		}
	}

	return true
}
```